lib/internal/linkedlist.js

```diff
@@ -42,10 +49,17 @@ exports.remove = remove;
 
 // remove a item from its list and place at the end.
 function append(list, item) {
-  remove(item);
+  if (item._idleNext || item._idlePrev) {
```

Comment:
Would the common case of no trailing list element make it better to check these in the opposite order?